data-products/src/main/scala/org/sunbird/analytics/job/report/CollectionSummaryJob.scala

```diff
@@ -0,0 +1,183 @@
+    "startdate" -> "Batch start date",
+    "enddate" -> "Batch end date",
+    "hasCertified" -> "Has certificate",
+    "enrolledUsersCountByState" -> "Total enrolments By State",
+    "completionUserCountByState" -> "Total completion By State",
+    "userstate" -> "State",
+    "certificateIssueCount" -> "Total Certificate issued by State"
+  )
+
+  // $COVERAGE-OFF$ Disabling scoverage for main and execute method
+  override def main(config: String)(implicit sc: Option[SparkContext] = None, fc: Option[FrameworkContext] = None) {
+    JobLogger.init(jobName)
+    JobLogger.start(s"$jobName started executing", Option(Map("config" -> config, "model" -> jobName)))
+    implicit val jobConfig: JobConfig = JSONUtils.deserialize[JobConfig](config)
+    implicit val spark: SparkSession = openSparkSession(jobConfig)
+    implicit val frameworkContext: FrameworkContext = getReportingFrameworkContext()
+    init()
+    val conf = config.split(";")
+    val batchIds = if (conf.length > 1) {
+      conf(1).split(",").toList
+    } else List()
+    try {
+      val res = CommonUtil.time(prepareReport(spark, fetchData, batchIds))
+      saveToBlob(res._2) // Saving report to blob stroage
+      JobLogger.end(s"$jobName completed execution", "SUCCESS", Option(Map("timeTaken" -> res._1, "totalRecords" -> res._2.count())))
+    } finally {
+      frameworkContext.closeContext()
+      spark.close()
+    }
+  }
+
+  // $COVERAGE-OFF$ Disabling scoverage for main and execute method
+  def init()(implicit spark: SparkSession, fc: FrameworkContext, config: JobConfig) {
+    DecryptUtil.initialise()
+    spark.setCassandraConf("UserCluster", CassandraConnectorConf.ConnectionHostParam.option(AppConf.getConfig("sunbird.user.cluster.host")))
+    spark.setCassandraConf("LMSCluster", CassandraConnectorConf.ConnectionHostParam.option(AppConf.getConfig("sunbird.courses.cluster.host")))
+    spark.setCassandraConf("ContentCluster", CassandraConnectorConf.ConnectionHostParam.option(AppConf.getConfig("sunbird.content.cluster.host")))
+  }
+
+  // $COVERAGE-ON$
+  def getUserData(spark: SparkSession, fetchData: (SparkSession, Map[String, String], String, StructType) => DataFrame): DataFrame = {
+    val schema = Encoders.product[UserData].schema
+    fetchData(spark, userCacheDBSettings, "org.apache.spark.sql.redis", schema)
+      .withColumn("username", concat_ws(" ", col("firstname"), col("lastname")))
+      .withColumn("userstate", when(col("state").isNotNull && col("state") =!= "", col("state")).otherwise("NA"))
+  }
+
+  def getUserEnrollment(spark: SparkSession, fetchData: (SparkSession, Map[String, String], String, StructType) => DataFrame): DataFrame = {
+    fetchData(spark, userEnrolmentDBSettings, cassandraUrl, new StructType())
+      .select(col("batchid"), col("userid"), col("courseid"), col("active")
+        , col("completionpercentage"), col("enrolleddate"), col("completedon"), col("status"), col("certificates"), col("issued_certificates"))
+  }
```

Comment:
Why are we selecting "active" and "completionpercentage" columns if we are going to drop them off?